vortexor/src/stake_updater.rs

```diff
@@ -0,0 +1,98 @@
+//! Module responsible for updating the staked key map.
+//! Adapted from jito-replayer code.
+
+use {
+    crate::load_balancer::RpcLoadBalancer,
+    log::warn,
+    solana_client::client_error,
+    solana_sdk::pubkey::Pubkey,
+    solana_streamer::streamer::StakedNodes,
+    std::{
+        collections::HashMap,
+        str::FromStr,
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc, RwLock,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::{Duration, Instant},
+    },
+};
+
+const PK_TO_STAKE_REFRESH_DURATION: Duration = Duration::from_secs(5);
+
+pub struct StakeUpdater {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl StakeUpdater {
+    pub fn new(
+        exit: Arc<AtomicBool>,
+        rpc_load_balancer: Arc<RpcLoadBalancer>,
+        shared_staked_nodes: Arc<RwLock<StakedNodes>>,
+        staked_nodes_overrides: HashMap<Pubkey, u64>,
+    ) -> Self {
+        let thread_hdl = Builder::new()
+            .name("stkUpdtr".to_string())
+            .spawn(move || {
+                let mut last_stakes = Instant::now();
+                while !exit.load(Ordering::Relaxed) {
+                    let mut stake_map = Arc::new(HashMap::new());
```

Comment:
This pattern is very weird.
`try_refresh_pk_to_stake` doesn't use this value, it overwrites it. Why not just have `try_refresh_pk_to_stake` return this instead of a `bool`?

OR have it re-use the allocation for building stake map. if passing it in, just pass a &mut HashMap, not an Arc.